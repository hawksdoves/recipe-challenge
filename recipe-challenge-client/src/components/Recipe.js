import React from 'react';
import { Card } from 'semantic-ui-react';

import Ingredient from './Ingredient';

function Recipe(props) {
    const { recipe } = props
    if (recipe) {
        return (
            <Card key={recipe.id} >
                <Card.Content>
                    <Card.Header>{recipe.name}</Card.Header>
                    <Card.Meta>{recipe.time} minutes</Card.Meta>
                    {recipe.ingredients.map(ingredient => {
                        return (
                            <Ingredient ingredient={ingredient} key={`${recipe.id}-${ingredient.name}`}/>
                        )
                    })}
                </Card.Content>
            </Card>
        )
    }
    return <div>Sorry, this recipe doesn't exist or may have been removed</div>
};

export default Recipe;