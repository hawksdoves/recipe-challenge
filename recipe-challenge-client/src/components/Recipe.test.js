import React from 'react';
import { shallow } from 'enzyme';

import Recipe from './Recipe';

describe('Recipe', () => {
    it('should render the recipe correctly', () =>{
        const recipe = {
            id: '1',
            ingredients: [
                {
                    name: 'steak',
                    quantity: '3'
                }
            ]
        }
        const wrapper = shallow(<Recipe recipe={recipe} />);

        expect(wrapper).toMatchSnapshot()
    })

    it('should render error message if recipe does not exist', () =>{
        const wrapper = shallow(<Recipe />);

        expect(wrapper).toMatchSnapshot()
    })
});
