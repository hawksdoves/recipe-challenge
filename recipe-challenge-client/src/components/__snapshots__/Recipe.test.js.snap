// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should render error message if recipe does not exist 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    Sorry, this recipe doesn't exist or may have been removed
</div>,
  "nodes": Array [
    <div>
      Sorry, this recipe doesn't exist or may have been removed
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Recipe />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          Sorry, this recipe doesn't exist or may have been removed
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          Sorry, this recipe doesn't exist or may have been removed
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Recipe />,
}
`;

exports[`Recipe should render the recipe correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card>
    <CardContent>
        <CardHeader />
        <CardMeta>
             minutes
        </CardMeta>
        <Ingredient
            ingredient={
                Object {
                    "name": "steak",
                    "quantity": "3",
                  }
            }
        />
    </CardContent>
</Card>,
  "nodes": Array [
    <Card>
      <CardContent>
            <CardHeader />
            <CardMeta>
                   minutes
            </CardMeta>
            <Ingredient
                  ingredient={
                        Object {
                              "name": "steak",
                              "quantity": "3",
                            }
                  }
            />
      </CardContent>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Recipe
        recipe={
                Object {
                        "id": "1",
                        "ingredients": Array [
                          Object {
                            "name": "steak",
                            "quantity": "3",
                          },
                        ],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipe": Object {
            "id": "1",
            "ingredients": Array [
              Object {
                "name": "steak",
                "quantity": "3",
              },
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card>
          <CardContent>
                    <CardHeader />
                    <CardMeta>
                               minutes
                    </CardMeta>
                    <Ingredient
                              ingredient={
                                        Object {
                                                  "name": "steak",
                                                  "quantity": "3",
                                                }
                              }
                    />
          </CardContent>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card>
          <CardContent>
                    <CardHeader />
                    <CardMeta>
                               minutes
                    </CardMeta>
                    <Ingredient
                              ingredient={
                                        Object {
                                                  "name": "steak",
                                                  "quantity": "3",
                                                }
                              }
                    />
          </CardContent>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Recipe
    recipe={
        Object {
            "id": "1",
            "ingredients": Array [
              Object {
                "name": "steak",
                "quantity": "3",
              },
            ],
          }
    }
/>,
}
`;
